language/compiler/ir-to-bytecode/syntax/src/spec_language_ast.rs

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::ast::{BinOp, CopyableVal, StructName, UnaryOp};
+use libra_types::account_address::AccountAddress;
+
+/// AST for the Move Prover specification language. Just postconditions for now
+
+/// A location that can store a value
+#[derive(PartialEq, Debug, Clone)]
+pub enum StorageLocation {
+    /// A formal of the current procedure
+    Formal(String),
+    /// A resource of type `type_` stored in global storage at `address`
+    GlobalResource {
+        type_: StructName,
+        address: Box<StorageLocation>,
+    },
+    /// An access path rooted at `base` with nonempty offsets in `fields`
+    AccessPath {
+        base: Box<StorageLocation>,
+        fields: Vec<String>,
+    },
+    /// Value of a storage location *after* the current procedure executes. Not applicable to
+    /// TxnSenderAddress. Address, or Return
+    Old(Box<StorageLocation>),
+    /// Sender address for the current transaction
+    TxnSenderAddress,
+    /// Account address constant
+    Address(AccountAddress),
+    /// The return value of the current procedure
+    Ret,
+    // TODO: & and * operators
+    // TODO: useful constants like U64_MAX
+    // TODO: add generics to GlobalResource
+}
+
+/// An expression in the specification languafge
+#[derive(PartialEq, Debug, Clone)]
+pub enum SpecExp {
+    /// A Move constant
+    Constant(CopyableVal),
+    /// A spec language storage location
+    StorageLocation(StorageLocation),
+    /// Lifting the Move exists operator to a storage location
+    GlobalExists {
+        type_: StructName,
+        address: StorageLocation,
+    },
+    /// Unary operators also suported by Move
```

Comment:
Does Move also has ==> and <==> as boolean operators? They are very important for specs.

Reply:
I agree they are very useful to have in the spec language.  If easy to add, please do it.  Is there a standard binding precedence for them?  Perhaps we can look at what Boogie does.

Reply:
No, these will have to be added separately. Agreed that they are important.

Reply:
I am planning to add them in a future PR; this one is meant to cover all parsing that doesn't involve thinking about precedence :).